vendor/bat-native-ads/src/bat/ads/internal/federated/log_entries/number_of_tabs_opened_in_past_30_minutes.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/federated/log_entries/number_of_tabs_opened_in_past_30_minutes.h"
+
+#include <cstdint>
+
+#include "bat/ads/ads_client.h"
+#include "bat/ads/internal/ads_client_helper.h"
+#include "bat/ads/internal/federated/covariate_logs_util.h"
+#include "bat/ads/internal/user_activity/user_activity.h"
+#include "bat/ads/pref_names.h"
+
+namespace ads {
+
+namespace {
+
+int64_t ComputeTabsOpened(const UserActivityEventList& events) {
+  const int64_t count = std::count_if(
+      events.cbegin(), events.cend(), [](const UserActivityEventInfo& event) {
+        return event.type == UserActivityEventType::kOpenedNewTab;
+      });
+
+  return count;
+}
+
+}  // namespace
+
+NumberOfTabsOpenedInPast30Minutes::NumberOfTabsOpenedInPast30Minutes() =
```

Comment:
should we make the number of minutes of the window an argument on initialisation of this log entry?

Reply:
good idea, but is it ok if we keep this for a future PR as the purpose of this PR is to introduce an end to end covariate log pipeline and not implement a final set of logs. Those will be added subsequently as we learn what works and what not.